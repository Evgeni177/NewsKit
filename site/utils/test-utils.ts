import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
