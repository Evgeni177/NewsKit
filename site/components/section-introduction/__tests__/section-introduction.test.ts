import {renderToFragmentWithTheme} from 'newskit/test/test-utils';
import {SectionIntroduction} from '..';

describe('SectionIntroduction', () => {
  test('renders with all props', () => {
    const props = {
      title: 'Title',
      children: 'Section Introduction',
    };
    const fragment = renderToFragmentWithTheme(SectionIntroduction, props);
    expect(fragment).toMatchSnapshot();
  });
});
