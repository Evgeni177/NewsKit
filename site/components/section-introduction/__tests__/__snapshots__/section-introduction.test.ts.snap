// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionIntroduction renders with all props 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

@media screen {
  .emotion-0 {
    padding: 0 8px;
    margin-top: 16px;
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    margin-left: 8.333333333333334%;
    -webkit-flex-basis: 66.66666666666667%;
    -ms-flex-preferred-size: 66.66666666666667%;
    flex-basis: 66.66666666666667%;
    max-width: 66.66666666666667%;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    padding: 0 12px;
    margin-top: 24px;
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
  }
}

.emotion-1 {
  margin-bottom: 24px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 0;
  background-color: transparent;
}

.emotion-4 {
  padding: 1px 0px;
}

@media screen and (max-width: 767px) {
  .emotion-4 {
    font-family: 'Poppins',sans-serif;
    font-size: 28px;
    line-height: 1.2857142857142858;
    font-weight: 700;
    letter-spacing: 0;
  }

  .emotion-4::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.18348214285714293em;
  }

  .emotion-4::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.4334821428571429em;
  }
}

@media screen and (min-width: 768px) {
  .emotion-4 {
    font-family: 'Poppins',sans-serif;
    font-size: 40px;
    line-height: 1.2;
    font-weight: 700;
    letter-spacing: 0;
  }

  .emotion-4::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.140625em;
  }

  .emotion-4::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.390625em;
  }
}

.emotion-5 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #162740;
}

.emotion-5 svg {
  fill: #162740;
}

.emotion-6 {
  margin-bottom: 40px;
}

.emotion-7 {
  margin: 0;
  padding: 1px 0;
  color: #555E74;
  font-family: 'Poppins',sans-serif;
  font-size: 20px;
  line-height: 1.6;
  font-weight: 500;
  letter-spacing: 0;
}

.emotion-7 svg {
  fill: #555E74;
}

.emotion-7::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.34062500000000007em;
}

.emotion-7::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.5906250000000001em;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <section
            class="emotion-4"
          >
            <h2
              class="nk-headline-heading emotion-5"
            >
              Title
            </h2>
          </section>
        </div>
      </div>
    </div>
    <div
      class="emotion-6"
    >
      <p
        class="emotion-7"
      >
        Section Introduction
      </p>
    </div>
  </div>
  .emotion-0 {
  box-sizing: border-box;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

@media screen {
  .emotion-0 {
    padding: 0 8px;
    margin-top: 16px;
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
  }
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    margin-left: 8.333333333333334%;
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-0 {
    padding: 0 12px;
    margin-top: 24px;
    -webkit-flex-basis: 66.66666666666667%;
    -ms-flex-preferred-size: 66.66666666666667%;
    flex-basis: 66.66666666666667%;
    max-width: 66.66666666666667%;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
